Feature: SF Audio amplifier mute - 2.9 edition

mha1 opened the pull request:
This PR adds "Audio amplifier mute" to the SF/GF list. Possible use cases:
- total silence, no noise, not only volume down
- home brew BT Audio module picking up the audio signal at the amplifier input to use wireless ear buds. Silencing the speakers is necessary. Using SF makes that easy

Eldenroot commented:
CZ:
`#define TR_SF_DISABLE_AUDIO_AMP        "Vypnutí zesilovače zvuku"`

ENG original text:
`#define TR_SF_DISABLE_AUDIO_AMP        "Audio Amp Off"`


@zyren CN / TW
 @Eldenroot CZ
 @HThuren DA
 @ParkerEde / @TheIsotopes DE
 @Pat6874 FR
 @offer-shmuely HE
 robustini IT
 @ToshihiroMakuuchi JP
 @ulfhedlund SE
 @ajjjjjjjj PL

HThuren commented:
DA
#define TR_SF_DISABLE_AUDIO_AMP        TR("Sluk audio amp", "Sluk audio amplifier")

Pat6874 commented:
#define TR_SF_DISABLE_AUDIO_AMP "Audio Amp Off"

FR

"Désact. Ampli Audio" 

Le 2 juil. 2023, 11:59, à 11:59, Eldenroot ***@***.***> a écrit:
>CZ:
>`#define TR_SF_DISABLE_AUDIO_AMP        "Vypnutí zesilovače zvuku"`
>
>ENG original text:
>`#define TR_SF_DISABLE_AUDIO_AMP        "Audio Amp Off"`
>
>
>@zyren CN / TW
> @Eldenroot CZ
> @HThuren DA
> @ParkerEde / @TheIsotopes DE
> @Pat6874 FR
> @offer-shmuely HE
> robustini IT
> @ToshihiroMakuuchi JP
> @ulfhedlund SE
> @ajjjjjjjj PL
>
>-- 
>Reply to this email directly or view it on GitHub:
>https://github.com/EdgeTX/edgetx/pull/3741#issuecomment-1616562992
>You are receiving this because you were mentioned.
>
>Message ID: ***@***.***>

mha1 commented:
I have been transaulted :-)

@elecpower: Companion done, I'd appreciate your review

- updated companion for new SF
- updated radio languages CZ, DA, FR
- updated companion languages CZ, DA, FR

elecpower commented on companion/src/firmwares/customfunctiondata.h:
```diff
@@ -71,6 +71,7 @@ enum AssignFunc {
   FuncBindExternalModule,
   FuncRacingMode,
   FuncDisableTouch,
+  FuncDisableAudioAmp,
```
New functions must be added at the end of list to avoid breaking binary import or enum cross references

mha1 replied:
done

elecpower commented on companion/src/firmwares/edgetx/yaml_customfunctiondata.cpp:
```diff
@@ -65,6 +65,7 @@ static const YamlLookupTable customFnLut = {
   {  FuncScreenshot, "SCREENSHOT"  },
   {  FuncRacingMode, "RACING_MODE"  },
   {  FuncDisableTouch, "DISABLE_TOUCH"  },
+  {  FuncDisableAudioAmp, "DISABLE_AUDIO_AMP"  },
```
For consistency with enum move to end of list

mha1 replied:
done

robustini commented:
IT
#define TR_SF_DISABLE_AUDIO_AMP TR("Amp. audio off", "Amplificatore audio off")

ParkerEde commented:
DE
#define TR_SF_DISABLE_AUDIO_AMP        "Audio Amp AUS"

raphaelcoeffic commented:
@ParkerEde has this already been implemented in `main`? Normally we implement new things in `main`, and back-port / cherry-pick if really necessary into a stable branch (ex: `2.9`), not the opposite.

zyren commented:
CN
```c
#define TR_SF_DISABLE_AUDIO_AMP        "关闭音频功放"
```
TW 
```c
#define TR_SF_DISABLE_AUDIO_AMP        "關閉音頻功放"
```

mha1 commented:
@raphaelcoeffic I know, but main was in flux and changing yaml datastructs logic. Find PR ported to main here: https://github.com/EdgeTX/edgetx/pull/3753

pfeerick commented:
The other thing you should have kept in mind is 2.9.0 has been essentially in feature freeze, and is now in RC, so basing this on 2.9 is more than a bit naughty ;)

mha1 commented:
what? why? me naughty? I plead total ignorance your honor.

ajjjjjjjj commented:
PL
```
#define TR_SF_DISABLE_AUDIO_AMP TR("Wycisz wzm.", "Wycisz wzmacniacz audio")
```

mha1 commented:
@pfeerick Seriously, say the word and I'll close it. Until then I'll keep it alive and updated.

ulfhedlund commented:
#define TR_SF_DISABLE_AUDIO_AMP        TR("Audioförst. av","Audioförstärkare av")

pfeerick commented:
> @pfeerick Seriously, say the word and I'll close it. Until then I'll keep it alive and updated.

Nah, keep it alive... I'm inclined to let this infraction slide 🤪. Might as well round that feature off a bit more since you've already done the 2.9 version.

pfeerick commented:
btw, I think you should be able to knock the `type="unfinished"` flag off the companion translations that are translated ;)

mha1 commented:
I accept the sentence as hard as it may be, your Honor

ToshihiroMakuuchi commented:
JP
#define TR_SF_DISABLE_AUDIO_AMP        "オーディオアンプ無効"

brainbubblersbest commented:
Ehm sorry. But why not for the X12S?
FROS mutes the Audio Ampflifier too when using external r9m to prevent the humming noise. 
So it does exactly the same on fros as the other radios that are supported here on Edgetx.

But Audio is Shutdown here all the time if no Audio file is played.

mha1 commented:
@3djc your word on this?

3djc commented:
To my knowledge, fros uses the x12s audio chip in a completely different way that allows them to mute, but would not work at all with EdgeTx

mha1 commented:
🙏

brainbubblersbest commented:
> To my knowledge, fros uses the x12s audio chip in a completely different way that allows them to mute, but would not work at all with EdgeTx (and the audio chip mute pin is soldered to the ground)

Yes it is soldered to Ground.
And Yes FROS uses a second Chip for mp3 decoding.
But that Chip was never used at all with opentx and edgetx.
The Shutdown Pin from

3djc commented:
If you are talking about PI.09, then I suggest you read this : https://github.com/opentx/opentx/blob/00d3d62f6a5acacc35e2c3f71608bd7ae06a839c/radio/src/targets/horus/audio_spi_driver.cpp#L362

brainbubblersbest commented:
Thats interesting and weird. The datasheet describes startup time on that
IC with 1,6us.
Thank you for that explanation.

Im still available for testing if you will give it a try. But u have a good
Argument against stuff in this direction. 😕


3djc ***@***.***> schrieb am Di., 22. Aug. 2023, 22:03:

> If you are talking about PI.09, then I suggest you read this :
> https://github.com/opentx/opentx/blob/00d3d62f6a5acacc35e2c3f71608bd7ae06a839c/radio/src/targets/horus/audio_spi_driver.cpp#L362
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/EdgeTX/edgetx/pull/3741#issuecomment-1688847249>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ALPVHWOYUZBSXOO2OUGTZWTXWUGA5ANCNFSM6AAAAAAZ25XSRY>
> .
> You are receiving this because you commented.Message ID:
> ***@***.***>
>

raphaelcoeffic commented:
@brainbubblersbest the real question is whether or not the `MUTE` pin (marked blue in your picture) is somehow wired to other things as well like shutdown. Regarding start-up time, it is specified in the datasheet as 1.2s (up to 2s):
<img width="525" alt="Screenshot 2023-08-23 at 07 42 17" src="https://github.com/EdgeTX/edgetx/assets/1050031/7d1129c2-be4e-41f3-ae7f-d6af40f42ee0">

That being said, if you guys want to reverse engineer how it is used in FrOS, go ahead! We can see how that can be implemented then.

That being said, the X12S audio circuit is fairly noise-free, so that it can be implemented in a different way, for instance by stopping data transfer to the `VS1053b`.

3djc commented:
> @brainbubblersbest the real question is whether or not the `MUTE` pin (marked blue in your picture) is somehow wired to other things as well like shutdown. 

I can address this one:
![image](https://github.com/EdgeTX/edgetx/assets/5167938/c16d3549-f269-4e06-b8e6-73d1be2cbf73)
